### DISCUSSION TITLE
Its possible to pass values between it blocks?

### DISCUSSION THREAD

**Discussion by Adolfo-Macias** [cypress-io/cypress#16034] [Questions and Help] [answered]
I have tried the alias  .as() https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html but the value it overwritten when when the next it block starts

I want to read a value in one it block and use that value in the 2nd it block

it('Read a value from the page', () => {
  // get the value in variable FOO
})

it('Use the value in variable FOO', () => {
  // Use the value got from previous it block
})

**Comment by bahmutov** [comment, marked as answer]
Aliases are reset before each test https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Aliases-are-reset-before-each-test so you should recreate them if you need them in both tests. If you simply need to pass values between the tests, use a local closure variable

```js
let foo
it('Read a value from the page', () => {
// get the value in variable FOO
   cy.get('...').invoke('text').then(t => foo = t)
})

it('Use the value in variable FOO', () => {
// Use the value got from previous it block
   expect(foo).to.equal(...)
})
```

**WARNING** this approach assumes the tests run in a specific order! We think this is a bad practice, the tests should be independent.

**Reply by Adolfo-Macias** [reply]
Thanks @bahmutov , I think this will solve my problem. Ill try it.
